src/basic/memstream-util.h

```diff
@@ -0,0 +1,13 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+#pragma once
+
+#include <stdio.h>
+
+#include "macro.h"
+
+typedef struct MemStream MemStream;
```

Comment:
hmm, wouldn't it make sense to just open code this? so that we can save the extra memory allocation for the structure itself?

i mean, hiding struct fields/encapsulating things is of course nice, but afaics we only will allocate these on the stack, and what's in it isn't really much of a surprise or secret, hence maybe make the definition of the struct public so that callers can just allocate it on the stack and we save the extra malloc. (this doesn't mean callers should directly access the fields, they still shouldn't...)